minishell.c

```diff
@@ -46,6 +46,11 @@ int main(void) // no need for arguments can use `getenv("PATH")`
 
 		// 1. stdin loop
 		line = readline("~$ ");
+		if(!line) {
+			printf("exit\n");
+			exit(0);
```

Comment:
should it be if no line then continue to the next loop? doesn't exit exit the program entirely?

Reply:
Should have put some comments!

This is to handle ctrl + d.

ctrl d sends an EOF signal, making the return value from readline =null.

In this case if readline is null then we want to exit the shell. 

This is the only time readline will be null, if input is empty readline won't be null just empty

From the man -  "If EOF is encountered while reading a line,
       and the line is empty, NULL is returned."